### PR TITLE
Communication fixup

### PR DESCRIPTION
**Pull request by reidbaker**
This changes our intents to use the localbroadcastmanager while still allowing global broadcast intents to work. If you have the time please pull down my branch and compile and run. I am really worried about elusive bugs. That said I have tested this with a few devices and it seems to work.

### CODE REVIEW

**Comment by theJenix** [comment]
Git says you renamed BroadcastIntent to LocalBroadcastIntent.  Do we no longer have a BroadcastIntent class?

**Comment by theJenix** [comment]
Also, after seeing it in code, I would request that we rename BroadcastRegistrar addAction to addLocalAction, and addGlobalAction back to addAction.  This would be more consistent with how we named the broadcast intent classes.  Afterwards, it's good to :pushpin:

File: SoundStream/src/com/lastcrusade/soundstream/CustomApp.java [hunk 1]
```diff
@@ -106,12 +92,13 @@ public void onAudioFocusChange(int focusChange) {
 
     @Override
     public void onTerminate() {
-        unregisterReceivers();
-        externalControlClient.unregister();
+        unregisterExtranlControlClient();
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/CustomApp.java]: I'm not sure where this method is, but can we refactor it to spell external correctly?

File: SoundStream/src/com/lastcrusade/soundstream/util/BroadcastRegistrar.java [hunk 2]
```diff
@@ -36,43 +37,74 @@
...
-    private Map<String, IBroadcastActionHandler> handlerMap =
+    private IntentFilter localFilter;
+    private IntentFilter globalFilter;
+    private Map<String, IBroadcastActionHandler> localHandlerMap =
             new HashMap<String, IBroadcastActionHandler>();
-    private BroadcastReceiver internalReceiver;
+    private Map<String, IBroadcastActionHandler> globalHandlerMap =
+            new HashMap<String, IBroadcastActionHandler>();
+
+    private BroadcastReceiver localReceiver;
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/util/BroadcastRegistrar.java:47]: can we add a comment somewhere to explain when to use local vs global?

File: SoundStream/src/com/lastcrusade/soundstream/CustomApp.java (hunk 1 above)
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/CustomApp.java]: I want to say this was moved (and fixed :smile:) when I fixed audio ducking. I'll double check when I get to my computer. 

ejohnson44 notifications@github.com wrote:

> @@ -106,12 +92,13 @@ public void onAudioFocusChange(int focusChange) {
> 
> ```
>  @Override
>  public void onTerminate() {
> ```
> -        unregisterReceivers();
> -        externalControlClient.unregister();
> -        unregisterExtranlControlClient();

I'm not sure where this method is, but can we refactor it to spell external correctly?

---

Reply to this email directly or view it on GitHub:
https://github.com/TheLastCrusade/SoundStream/pull/151/files#r4278061

File: SoundStream/src/com/lastcrusade/soundstream/util/BroadcastRegistrar.java (hunk 2 above)
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/util/BroadcastRegistrar.java:47]: Yes I can

**Comment by reidbaker** [comment]
As of right now I think I have address all issues with this branch.

**Comment by theJenix** [comment]
This is ready to go, but we can't automatically merge this pull request.  @reidbaker, help us help you.

**Comment by reidbaker** [comment]
It is now possible to merge into master.

**Comment by theJenix** [comment]
:aerial_tramway:

**Pull request merged**